### ISSUE TITLE
When promoting app from stage to run, dialog is displayed for fraction of a second

### ISSUE THREAD

**Issue by ldimaggi** [openshiftio/openshift.io#4017]
Steps to recreate:
* Create new app in Openshift.io launcher
* Press "input required" button to deploy app to run
* Observe small dialog displayed for fraction of a second:
![frame-022](https://user-images.githubusercontent.com/642621/42845198-2eadd592-89e3-11e8-90bd-6a74e58fabe7.jpg)

* After which it is overwritten by the promote dialog:
![frame-028](https://user-images.githubusercontent.com/642621/42845223-4542072e-89e3-11e8-8be7-5b566cb0c560.jpg)

![myimage](https://user-images.githubusercontent.com/642621/42845314-87738c94-89e3-11e8-8a15-6c08d01a08b8.gif)

**Comment by hrishin** [comment]
@ldimaggi this is happening because dialog box checks if Jenkins is running and not idle before eanbling those buttons. What improvement are we expecting here? Better UX?

**Comment by ldimaggi** [comment]
Can we hide that additional dialog? It's confusing to see it displayed for only a fraction of a second.

**Comment by piyush1594** [comment]
@ldimaggi If Jenkins is in the idle state, then it will be there for a long time until it does not get unidle. This box was introduced for that purpose only.

**Comment by ldimaggi** [comment]
So - if Jenkins is not idle - then the dialog is only displayed very quickly, correct? Is it possible to check if Jenkins is idle, and if it is not, then not display the dialog?

**Comment by hrishin** [comment]
@ldimaggi  
> Jenkins is idle, and if it is not, then not display the dialog?

Sorry didnt get that.

**Comment by hrishin** [comment]
> Jenkins is idle, and if it is not, then not display the dialog?

@ldimaggi I think this dialogue is needed, however, we can try to enhance the overall UX around it.
i.e. UI can keep the active state of Jenkins (idle or not idle) by polling it or using watch request on Jenkins deployment. In that way, UI would not poll Jenkins each and every time while opening the dialogue box.

@joshuawilson @ldimaggi are we planning any enhancements here?

**Comment by joshuawilson** [comment]
Eng doesn't have any current plans to change this. However, if @openshiftio/uxd-team provides direction here we can add it to the backlog.

**Comment by piyush-garg** [comment]
@joshuawilson @ldimaggi @hrishin 

This seems like a UI issue and needs recommendations from the UX team. Nothing build team can do with this. Removing build team label and adding UX label

cc @sthaha @muruGanesan @pradeepto

**Comment by muruGanesan** [comment]
@piyush-garg, @joshuawilson @ldimaggi @hrishin,
The 'gif' animation (attachment) appears like an experience issue.
But when I tested with my application (pipeline), I found a huge delay in activating Jenkins. so I don't get a chance to see the second pop-up at all.

I suggest two corrections:
1) **Visual interference:** pop-up window's height changes abruptly - this can be fixed immediately.

2) Technical & experience issue: 
we (UX) team should discuss with the relevant engineering team and decide on an appropriate solution.

cc @sthaha @pradeepto, @serenamarie125

**Comment by joshuawilson** [comment]
I would agree that making them the same size makes sense. Please let us know what else should be changed.

**Comment by joshuawilson** [comment]
Promote and reject are actions against Jenkins. If Jenkins isn't there and you click the button it should fail with an error code that indicates Jenkins is down. There is no way to click those, have them work and not call Jenkins.

**Comment by muruGanesan** [comment]
@joshuawilson, thanks.
I too discussed with  @hrishin about the same point. [It is difficult to separate UI from the Jenkins status]
Considering the above scenario, we can do only the visual enhancement i.e. height, width, and the display position of both pop-ups/dialog boxes should be same.
